### PR TITLE
Virtual Assistant Client for Android - initial public commit

### LINKED ISSUE #1396: [Android App] VA response rendering
Enable vertical scrolling of a conversation history of the activities and cards sent from a Virtual Assistant

### LINKED ISSUE #1397: [Android App] Adaptive Card renderer
Use latest card renderer (`1.2.0`) to support the newest Adaptive Card features

### LINKED ISSUE #1441: [Android App] UI of dialog needs to have responses in bubbles as a configuration item
Add responses to be in bubbles and the color configurable.

### LINKED ISSUE #1445: [Android App] Application needs to be able to register as the default assistant on Android
Application should be able to be registered as the default assistant by the user on android.

### PR DESCRIPTION
**Pull request by abiemann**
## Description
Close #1445 
Close #1441 
Close #1396 
Close #1397 

## Testing Steps
Be sure to replace the DefaultConfiguration.java with one that contains the necessary keys and ID's

## Checklist

- [X ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the appropriate unit tests
- [ ] I have tested any new/updated dialogs using Speech in the emulator to ensure the [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) property is set to enable a high quality speech-first experience and the appropriate [InputHints](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) are set correctly. 
- [ ] I have updated related documentation

- [ ] A duplicate issue is filed to track future  work.

- [ ] All languages have been updated
- [ ] You have tested deployment with your new models

### CODE REVIEW

File: solutions/android/VirtualAssistantClient/app/src/main/assets/webchat.html [hunk 1]
```diff
@@ -0,0 +1,353 @@
+﻿<!DOCTYPE html>
```
Comment (darrenj) [inline solutions/android/VirtualAssistantClient/app/src/main/assets/webchat.html:1]: Is this needed with the android app?

File: solutions/android/VirtualAssistantClient/app/src/main/assets/webchat.html [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+        this.emitter.on(name, listener);
+      }
+
+      removeEventListener(name, listener) {
+        this.emitter.removeListener(name, listener);
+      }
+
+      connect() {
+        if (typeof window.CSharp === 'undefined') {
+          console.warn('Not running in Xamarin, cannot call Connect().');
```
Comment (darrenj) [inline solutions/android/VirtualAssistantClient/app/src/main/assets/webchat.html:72]: Xamarin?

File: solutions/android/VirtualAssistantClient/app/src/main/java/com/microsoft/bot/builder/solutions/virtualassistant/service/SpeechService.java [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+                    updateBotResponseWidget(botConnectorActivity.getText());
+                    break;
+                case "dialogState":
+                    Log.i(TAG_FOREGROUND_SERVICE, "Activity with DialogState");
+                    break;
+                case "PlayLocalFile":
+                    Log.i(TAG_FOREGROUND_SERVICE, "Activity with PlayLocalFile");
+                    playMediaStream(botConnectorActivity.getFile());
+                    break;
+                case "AbsoluteTemp": {
```
Comment (darrenj) [inline solutions/android/VirtualAssistantClient/app/src/main/java/com/microsoft/bot/builder/solutions/virtualassistant/service/SpeechService.java:276]: We should update these with the automotive skill event name formats (which we can change if needed). https://github.com/microsoft/botframework-solutions/blob/master/skills/src/csharp/automotiveskill/automotiveskill/Dialogs/VehicleSettingsDialog.cs#L520 

Shows how we send events back and this has an example. https://github.com/microsoft/botframework-solutions/blob/master/docs/reference/skills/automotive.md#event-responses

@ryanlengel can add the auto skill to help your testing

File: solutions/android/VirtualAssistantClient/app/src/main/java/com/microsoft/bot/builder/solutions/virtualassistant/service/SpeechService.java [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+        if (activityReceived.botConnectorActivity != null) {
+            BotConnectorActivity botConnectorActivity = activityReceived.botConnectorActivity;
+
+            String amount;
+
+            switch (botConnectorActivity.getType()) {
+                case "message":
+                    updateBotResponseWidget(botConnectorActivity.getText());
+                    break;
+                case "dialogState":
```
Comment (darrenj) [inline solutions/android/VirtualAssistantClient/app/src/main/java/com/microsoft/bot/builder/solutions/virtualassistant/service/SpeechService.java:269]: Should we not also have the poi event hooked up here too and fire navigation deep link

Also see this event which gives you a number to dial when joining a meeting: https://github.com/microsoft/botframework-solutions/blob/master/skills/src/csharp/calendarskill/calendarskill/Dialogs/ConnectToMeetingDialog.cs#L404

File: solutions/android/VirtualAssistantClient/app/src/main/res/xml/assistant_service.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<voice-interaction-service
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:recognitionService="com.microsoft.bot.builder.solutions.virtualassistant.assistant.AssistantService"
+    android:sessionService="com.microsoft.bot.builder.solutions.virtualassistant.assistant.AssistantSessionService"
+    android:supportsAssist="true"/>
+
+    <!--
```
Comment (darrenj) [inline solutions/android/VirtualAssistantClient/app/src/main/res/xml/assistant_service.xml:8]: Remove?

**Review by darrenj — CHANGES_REQUESTED** [review]
Nothing major, would be good to align with event names with auto, poi and calendar skill, we can take the opportunity to better normalise them too so they are more consistent across skills?

File: solutions/android/VirtualAssistantClient/app/src/main/assets/webchat.html (hunk 1 above)
Comment (abiemann) [reply solutions/android/VirtualAssistantClient/app/src/main/assets/webchat.html:1]: I agree that the webchat interface could be removed - it was used early on for basic testing

File: solutions/android/VirtualAssistantClient/app/src/main/java/com/microsoft/bot/builder/solutions/virtualassistant/service/SpeechService.java (hunk 3 above)
Comment (abiemann) [reply solutions/android/VirtualAssistantClient/app/src/main/java/com/microsoft/bot/builder/solutions/virtualassistant/service/SpeechService.java:276]: dewain didn't want any automotive things in this app. What you found was some automotive code that I should have removed - it sent info to automotive widgets that I had already removed

File: solutions/android/VirtualAssistantClient/app/src/main/res/xml/assistant_service.xml (hunk 5 above)
Comment (abiemann) [reply solutions/android/VirtualAssistantClient/app/src/main/res/xml/assistant_service.xml:8]: no that's intentional - it ties together classes to let the user register the app as an assistant.

File: solutions/android/VirtualAssistantClient/app/src/main/java/com/microsoft/bot/builder/solutions/virtualassistant/service/SpeechService.java (hunk 4 above)
Comment (abiemann) [reply solutions/android/VirtualAssistantClient/app/src/main/java/com/microsoft/bot/builder/solutions/virtualassistant/service/SpeechService.java:269]: I'll write a ticket for this feedback to add after KWS

**Comment by ryanisgrig** [comment]
Tracking the events at #1393

**Review by ryanisgrig — APPROVED** [review]

**Review by darrenj — APPROVED** [review]

**Pull request merged**